DataverseGen.Cli/Program.cs

```diff
@@ -61,17 +62,29 @@ private static void Main(string[] args)
                 DataverseConnector connector = new DataverseConnector(config.ConnectionString, config.Entities);
                 MappingEntity[] data = connector.GetMappedEntities();
                 Console.WriteLine("Finish Load data");
-                // Generator gen = new Generator("Ttfile.tt","out\\",null);
-
-                //  gen.GenerateTemplate();
                 Context context = new Context
                 {
                     Namespace = config.Namespace,
                     Entities = data
                 };
+
                 Console.WriteLine("Start generator");
-                Generator gen2 = new Generator("dataversetemplate.tt", config.OutDirectory, context);
-                gen2.GenerateTemplate();
+                switch (config.TemplateEngine)
```

Comment:
you could go for a common interface with "GenerateTemplate" as a contract. Then within the switch there would be simple assignments or even this could be extracted to factory